toqito/matrices/gen_pauli_x.py

```diff
@@ -51,9 +51,4 @@ def gen_pauli_x(dim: int) -> np.ndarray:
     :return: :code:`dim`-by-:code:`dim` gen_pauli_x matrix.
 
     """
-    gen_pauli_x_mat = np.identity(dim)
-    gen_pauli_x_mat = np.roll(gen_pauli_x_mat, -1)
-    gen_pauli_x_mat[:, -1] = np.array([0] * dim)
-    gen_pauli_x_mat[0, -1] = 1
-
-    return gen_pauli_x_mat
+    return np.roll(np.identity(dim), -1, axis=1)
```

Comment:
```suggestion
    return np.roll(np.identity(dim), -1, axis=1)
    # first column of the identity matrix becomes the last column due to `shift = -1` and `axis=1`
    # https://numpy.org/doc/stable/reference/generated/numpy.roll.html
```

Reply:
I think I'd opt to not have comments here as the call to `roll` based on the existing `numpy` docs is clear enough (at least in my mind). Or, if we are to include a comment, I would just include 

```py
# First column of the identity matrix becomes the last column due to `shift = -1` and `axis=1`
```

but place it above the line

```py
    return np.roll(np.identity(dim), -1, axis=1)
```

Either moving the comment or not including it work for me!